### ISSUE TITLE
Index Checker cannot compare `@LessThan("2+1")` against `@LessThan("3")`

### ISSUE THREAD

**Issue by mernst** [typetools/checker-framework#2541]
The Index Checker issues a false positive warning for the following code.

```
public class LessThanConstantAddition {

    public static void checkedPow(int b) {
        if (b <= 2) {
            int c = (int) b;
        }
    }
}
```
Its output is:
```
javac -processor index LessThanConstantAddition.java
LessThanConstantAddition.java:5: warning: [cast.unsafe] cast from "@LessThan("2 + 1") int" to "@LessThan("3") int" cannot be statically verified
            int c = (int) b;
                    ^
1 warning
```

This comes up in Guava.
